src/services/auth.service.ts

```diff
@@ -24,12 +24,16 @@ class AuthService extends Repository<UserEntity> {
     return `Authorization=${tokenData.token}; HttpOnly; Max-Age=${tokenData.expiresIn};`;
   }
 
-  public async register({ password, ...userData }: RegisterUserRequestDto): Promise<BaseUserResponseDTO> {
+  public async register({ password, ...userData }: RegisterUserRequestDto): Promise<{ cookie: string; data: BaseUserResponseDTO }> {
     const foundUser: BaseUserResponseDTO = await UserEntity.findOne({ where: { email: userData.email } });
     if (foundUser) throw new HttpException(409, `This email ${userData.email} already exists`);
     const hashedPassword = await hash(password, 10);
-    const createUserData: BaseUserResponseDTO = await UserEntity.create({ ...userData, password: hashedPassword }).save();
-    return createUserData;
+    // Extract out the password from the response
```

Comment:
This is to extract password from the response.
Thinking we should talk about a better way of handling situations like this.
Maybe exclude password from every query and then explicility select it when we do need it?

Reply:
I think I read something about the Entity and setting Column({ select : false}) to prevent the password from being accessible